index.js

```diff
@@ -75,8 +75,12 @@ function resolveImpl(root, config) {
         return resolveImpl(root, { name: config });
     }
 
-    if (!config || !config.name) {
-        throw new TypeError('Module not defined.');
+    if (!config) {
+        throw new TypeError("No module section given in middleware entry");
+    }
+
+    if (!config.name) {
```

Comment:
hm? This is in the error path, so flip that: if there's no config at all, or just no name, error.